api/proto/docker_registry.proto

```diff
@@ -0,0 +1,94 @@
+
+    // The name of the primary account.
+    string primaryAccount = 3;
+}
+
+// A credential able to authenticate against a set of Docker repositories.
+message DockerRegistryAccount {
+    // The name of the account.
+    string name = 1;
+
+    // (Required) The registry address from which to pull and deploy images
+    // (e.g., `index.docker.io`).
+    string address = 2;
+
+    // The number of seconds between polling the Docker registry. Certain
+    // registries are sensitive to over-polling, and larger intervals
+    // (e.g., 10 minutes = 600 seconds) are desirable if you experience rate
+    // limiting. Defaults to `30`.
+    int32 cacheIntervalSeconds = 3;
+
+    // The number of threads on which to cache all provided repositories.
+    // Really only useful if you have a ton of repos. Defaults to 1.
+    int32 cacheThreads = 4;
+
+    // Timeout in milliseconds for provided repositories. Defaults to
+    // `60,000`.
+    int32 clientTimeoutMillis = 5;
+
+    // The email associated with your Docker registry. Often this only needs to
+    // be well-formed, rather than be a real address.
+    string email = 6;
+
+    // The environment name for the account. Many accounts can share the
+    // same environment (e.g., dev, test, prod).
+    string environment = 7;
+
+    // If `true`, Spinnaker will treat the Docker registry as insecure and not
+    // validate the SSL certificate. Defaults to `false`.
+    bool insecureRegistry = 8;
+
+    // Pagination size for the Docker `repository _catalog` endpoint. Defaults
+    // to `100`.
+    int32 paginateSize = 9;
+
+    // The Docker registry password. Only one of `password`, `passwordCommand`,
+    // and `passwordFile` should be specified.
+    string password = 10;
+
+    // Command to retrieve Docker token/password. The command must be available
+    // in the environment. Only one of `password`, `passwordCommand`, and
+    // `passwordFile` should be specified.
+    string passwordCommand = 11;
+
+    // The path to a file containing your Docker password in plaintext (not a
+    // Docker `config.json` file). Only one of `password`, `passwordCommand`,
+    // and `passwordFile` should be specified.
+    string passwordFile = 12;
+
+    // Fiat permissions configuration.
+    Permissions permissions = 13;
+
+    // (Deprecated) List of required Fiat permission groups. Configure
+    // `permissions` instead.
+    repeated string requiredGroupMemberships = 14;
+
+    // An optional list of repositories from which to cache images. If not
+    // provided, Spinnaker will attempt to read accessible repositories from the
+    // `registries _catalog` endpoint.
+    repeated string repositories = 15;
+
+    // If `true`, Spinnaker will sort tags by creation date. Defaults to
+    // `false`.
+    bool sortTagsByDate = 16;
+
+    // If `true`, Spinnaker will track digest changes. This is not recommended
+    // because it consumes a high QPM, and most registries are flaky.
```

Comment:
Is "QPM" here queries per minute?  Maybe this could be replaced by a more generic term (ie, not specific to per minute) such as "because it greatly increases queries to the registry".

Reply:
I was also not sure what QPM meant and assumed it was a super common abbrev for "queries per minute" that I had just never heard of...but I like your wording better; updated!